src/datadog/search/grammar.rs

```diff
@@ -33,32 +46,28 @@ impl QueryVisitor {
 
     fn visit_query(token: Pair<Rule>, default_field: &str) -> QueryNode {
         let contents = token.into_inner();
-        let mut clauses: Vec<LuceneClause> = Vec::new();
-        let mut modifier: Option<LuceneOccur> = None;
+        let mut is_not: bool = false;
+
+        //  AND takes precedence over OR.
+        // We will combine each consecutive clause in an AND group,
+        // and create a new and_group every time we encounter an OR.
+        // Finally, we will combine all the and_groups with OR.
+
+        let mut and_groups: Vec<QueryNode> = Vec::new();
+
+        let mut and_group: Vec<QueryNode> = Vec::new();
+
         for node in contents {
-            let clause: Option<LuceneClause> = match node.as_rule() {
+            let query_node: Option<QueryNode> = match node.as_rule() {
                 Rule::multiterm => Some(Self::visit_multiterm(node, default_field)),
```

Comment:
Do we also need the same "and" logic handling with the implicit "and" joins between the parts of a multiterm?